rfcs/0005-pablo-distribution-assets/0005-pablo-distribution.adoc

```diff
@@ -540,6 +534,28 @@ endwhile (no)
 stop
 ----
 
+==== Claim
+
+[plantuml,images/claim,png]
+----
+start
+->assetId\npositionId;
+:position = readStorage(positionId);
+:rewards_pool = readStorage(assetId);
+:penalty = reward_pool.lock.unlock_penalty;
+while (reward = rewards_pool.rewards.has_next) is (yes)
+    :claim = reward.total_rewards * position.shares /\nrewards_pool.total_shares - position.reductions[reward.asset_id];
+    note right: as per section 5.3.1.2
+    :claim = min(claim * (1 - penalty), reward.total_rewards - reward.claimed_rewards);
```

Comment:
We won't apply a penalty for just claiming

Reply:
b53537bfbe37aa2fd87ad227de696d55508a12e9